src/utils/screenshot.util.ts

```diff
@@ -0,0 +1,56 @@
+import {PNG, PNGWithMetadata} from 'pngjs';
+import * as fs from 'fs';
+import Pixelmatch from 'pixelmatch';
+
+import {ElementFinder} from "../wdio";
+import {ReportUtility} from "./report.util";
+import {AssertionUtility} from "./assertion.util";
+import {Widget} from "../widgets";
+import {BasePage} from "../pages";
+
+
+export class ScreenshotUtility {
+    private static basePath = './screenshots';
+    private static pixelTolerance = 0;
+
+    public static setBasePath(basePath: string): void {
+        this.basePath = basePath;
+    }
+
+    public static setPixelThreshold(pixelTolerance: number): void {
+        this.pixelTolerance = pixelTolerance;
+    }
+
+    public static async expectPageScreenshot(page: BasePage, fileName: string, pageDescriptiveName: string): Promise<void> {
+        return this.expectScreenshot(page.getElementFinder(), fileName, pageDescriptiveName);
+    }
+
+    public static async expectWidgetScreenshot(widget: Widget, fileName: string, widgetDescriptiveName: string): Promise<void> {
+        return this.expectScreenshot(widget.getElement(), fileName, widgetDescriptiveName);
+    }
+
+    public static async expectScreenshot(element: ElementFinder, fileName: string, elementDescriptiveName: string): Promise<void> {
+        const actualImageFilepath = `${this.basePath}/${fileName}.actual.png`;
+        await element.saveScreenshot(actualImageFilepath);
+
+        const expectedImageFilepath = `${this.basePath}/${fileName}.expected.png`;
+        await ReportUtility.addExpectedResult(`'${elementDescriptiveName}' is equal to screenshot of '${fileName}.expected.png' file`, async () => {
+            const diffImageFilepath = `${this.basePath}/${fileName}.diff.png`;
+            const diffPixels = this.compareImageFiles(expectedImageFilepath, actualImageFilepath, diffImageFilepath);
+            AssertionUtility.expectLowerOrEqual(diffPixels, this.pixelTolerance);
+        });
+    }
+
+    private static compareImageFiles(expectedImageFilepath: string, actualImageFilepath: string, diffImageFilepath: string): number {
+        const expectedImage: PNGWithMetadata = PNG.sync.read(fs.readFileSync(expectedImageFilepath));
+        const actualImage: PNGWithMetadata = PNG.sync.read(fs.readFileSync(actualImageFilepath));
+
+        const { width, height } = expectedImage;
+        const diffImage = new PNG({width, height});
+        const numDiffPixels = Pixelmatch(expectedImage.data, actualImage.data, diffImage.data, width, height, {threshold: 0.1});
```

Comment:
Maybe it would be worth to have a new configurable property for this threshold, so it can be changed from the consumer side in case the 0.1 value is not matching the expectations.

Reply:
I would suggest using `imageMatchingThreshold` as a name for this new parameter.

Reply:
Added a configuration property for this attribute with is corresponding setter, will be present in next commit